src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/GeneralizedCoastlineCheck.java

```diff
@@ -121,4 +123,78 @@ private double getGeneralizedSegmentPercentage(final LineItem line)
                 .collect(Collectors.toList());
     }
 
+    /**
+     * This method validates if the relation members have the tag natural=coastline
+     *
+     * @param object
+     *            The {@link AtlasObject} being checked.
+     * @return true if the {@link AtlasObject} has the tag natural=coastline
+     */
+    private boolean hasCoastlineRelationMembers(final AtlasObject object)
+    {
+        final Predicate<Relation> relationPredicate = relation -> isCoastline(relation);
+        return hasRelationMembers((LineItem) object, relationPredicate);
+    }
+
+    /**
+     * This method checks if a {@link LineItem} has relation members and satisfies the predicate.
+     *
+     * @param object
+     *            The {@link LineItem} being checked.
+     * @param relationPredicate
+     *            Predicate used to filter the relation.
+     * @return true if the {@link LineItem} has relation members and satisfies the predicate.
+     */
+    private boolean hasRelationMembers(final LineItem object,
+            final Predicate<Relation> relationPredicate)
+    {
+        return object.relations().stream().anyMatch(relationPredicate);
+    }
+
+    /**
+     * This method validates if the relation members have the tag source=PGS.
+     *
+     * @param object
+     *            The {@link AtlasObject} being checked.
+     * @return true if the {@link AtlasObject} has the tag source=PGS.
+     */
+    private boolean hasSourcePGSRelationMembers(final AtlasObject object)
+    {
+        final Predicate<Relation> relationPredicate = relation -> isSourcePGS(relation);
+        return hasRelationMembers((LineItem) object, relationPredicate);
+    }
+
+    /**
+     * This method checks if the AtlasObject has the tag natural=coastline.
+     *
+     * @param object
+     *            The {@link AtlasObject} being checked
+     * @return true if the {@link AtlasObject} has the tag natural=coastline.
+     */
+    private boolean isCoastline(final AtlasObject object)
+    {
+        return Validators.isOfType(object, NaturalTag.class, NaturalTag.COASTLINE);
+    }
+
+    /**
+     * This method checks if the {@link AtlasObject} has the tag source=PGS. If PGSfilter is set to
+     * false then this will always return true.
+     *
+     * @param object
+     *            The {@link AtlasObject} being checked
+     * @return true if the PGSfilter is set to true and if the {@link AtlasObject} has the tag
+     *         source=PGS. If the PGSfilter is set to false then this always returns true.
+     */
+    private boolean isSourcePGS(final AtlasObject object)
```

Comment:
Similar to my last comment, we could just have this method return true if the tag "source=pgs" exists without regard to the PGSfilter